app/services/action_queue.rb

```diff
@@ -24,20 +24,58 @@ def member
       @member ||= @action.member
     end
 
+
     class_methods do
       def push(action)
         new(action).push
       end
     end
   end
 
+  module Donatable
+    def user_data
+      {
+          first_name: member.first_name,
+          last_name:  member.last_name,
+          email:      member.email,
+          country:    country(member.country),
+          akid:       data[:akid],
+          postal:     data[:postal],
+          address1:   data[:address1],
+          source:     data[:source]
+      }.merge(UserLanguageISO.for(page.language) )
+    end
+
+    def data
+      @action.form_data.symbolize_keys
+    end
+
+    def extra_fields_for_subscription
+      {}.tap do |action|
+        if @action.form_data['is_subscription']
+          action[:skip_confirmation] = 1 if @action.form_data['recurrence_number'].to_i > 0
+          action[:fields] = action_fields
+        end
+      end
+    end
+  end
+
   class Pusher
     def self.push(action)
-      action.donation? ? DonationAction.push(action) : PetitionAction.push(action)
+      if action.donation
+        if action.form_data.fetch('payment_provider', '').inquiry.go_cardless?
+          DirectDebitAction.push(action)
+        else
+          DonationAction.push(action)
+        end
+      else
+        PetitionAction.push(action)
+      end
     end
   end
 
   class PetitionAction
+    include Donatable
```

Comment:
Why is a PetitionAction Donatable? Looking at Donatable, it seems like that just means that Donatable is misnamed, right?